controllers/openstackcluster_controller.go

```diff
@@ -181,6 +181,10 @@ func reconcileDelete(ctx context.Context, log logr.Logger, client client.Client,
 	}
 
 	// Delete other things
+	if err = networkingService.DeleteOrphanedPorts(); err != nil {
+		return reconcile.Result{}, errors.Errorf("failed to delete orophaned ports: %v", err)
+	}
+
 	if workerSecGroup := openStackCluster.Status.WorkerSecurityGroup; workerSecGroup != nil {
```

Comment:
This should fix: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/723